hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoApproveAllowanceHandler.java

```diff
@@ -396,10 +396,15 @@ public void updateSpender(
             final var nft = uniqueTokenStore.get(tokenId, serialNum);
             final var token = tokenStore.get(tokenId);
 
-            validateTrue(isValidOwner(nft, owner.accountNumber(), token), SENDER_DOES_NOT_OWN_NFT_SERIAL_NO);
-            final var copy = nft.copyBuilder()
-                    .spenderNumber(spenderId.accountNumOrThrow())
+            // FUTURE: owner will be updated to use an AccountID instead of a long account number
+            // Issue #7243
+            AccountID accountOwner = AccountID.newBuilder()
+                    .accountNum(owner.accountNumber())
+                    .realmNum(0)
```

Comment:
Suggestion:
Pull shard and realm from the payer account ID in the TransactionID, so we aren't hard-coding a value that has to be changeable.

Reply:
Owner Account does not have the ID yet. That's what issue #7243 is supposed to fix. When #7243 is implemented, this code should be updated to use that. #7243 is probably the next issue I will work on after this, so this hack of putting 0 in for realm and shard is very temporary.